test/e2e/suite/issuers/vault/certificate/cert_auth.go

```diff
@@ -273,7 +273,13 @@ func runVaultClientCertAuthTest(issuerKind string, testWithRoot bool, unsupporte
 			Expect(err).NotTo(HaveOccurred())
 
 			By("Validating the issued Certificate...")
-			err = f.Helper().ValidateCertificate(cert, validation.CertificateSetForUnsupportedFeatureSet(unsupportedFeatures)...)
+			err = f.Helper().ValidateCertificate(cert, validation.CertificateSetForUnsupportedFeatureSet(
+				// X509 certificate duration will not match requested duration in Certificate for
+				// all test cases. Instead, we disable the duration check here and compare the duration
+				// with our expected duration below (by calling ExpectDuration explicitly).
+				unsupportedFeatures.Clone().
+					Insert(featureset.DurationFeature),
+			)...)
 			Expect(err).NotTo(HaveOccurred())
```

Comment:
I find this very confusing. We call ValidateCertificate twice...first time we disable the duration check and then we call it again with a fixed expected duration. 

Why can't you create a list of validations, without the default duration check, and insert the fixed duration check.